packages/server/src/internals/procedure.ts

```diff
@@ -121,6 +114,20 @@ export class Procedure<TInputContext, TContext, TInput, TParsedInput, TOutput> {
     }
   }
 
+  private async parseOutput(rawOutput: unknown): Promise<TOutput> {
+    try {
+      if (process.env.NODE_ENV === 'production' || !this.parseOutputFn) {
+        return rawOutput as TOutput;
+      }
+      return await this.parseOutputFn(rawOutput);
+    } catch (cause) {
+      throw new TRPCError({
+        code: 'INTERNAL_SERVER_ERROR',
+        cause,
+      });
```

Comment:
Definitely, it's an internal server error if this is wrong. Unsure if we should do more to guide the developer.

Maybe something like this?:

```suggestion
      throw new TRPCError({
        code: 'INTERNAL_SERVER_ERROR',
        cause,
        message: 'Output validation failed',
      });
```

Reply:
done 👍